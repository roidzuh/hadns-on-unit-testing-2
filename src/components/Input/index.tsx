import React from 'react';
import { Props } from './types.type';

const Input = (props: Props) => {
  const { label, type } = props;
  return (
    <div>
      <label htmlFor={label}>{label}</label>
      {type === 'textarea' ? (
        <textarea name={label} id={label} />
      ) : (
        <input type={type} name={label} id={label} />
      )}
    </div>
  );
};

export default Input;
