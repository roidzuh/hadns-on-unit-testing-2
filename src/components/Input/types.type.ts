export type Props = {
  label: string;
  type: string;
};
