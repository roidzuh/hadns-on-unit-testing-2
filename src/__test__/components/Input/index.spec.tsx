import Input from '@/components/Input';
import { render } from '@testing-library/react';

describe('input component', () => {
  it('should handle render input textarea', () => {
    const { container } = render(<Input type="textarea" label="test" />);
    expect(container).toMatchSnapshot();
  });

  it('should handle render input text', () => {
    const { container } = render(<Input type="text" label="test" />);
    expect(container).toMatchSnapshot();
  });
});
