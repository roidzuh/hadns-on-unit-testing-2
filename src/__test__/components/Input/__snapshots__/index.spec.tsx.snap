// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input component should handle render input text 1`] = `
<div>
  <div>
    <label
      for="test"
    >
      test
    </label>
    <input
      id="test"
      name="test"
      type="text"
    />
  </div>
</div>
`;

exports[`input component should handle render input textarea 1`] = `
<div>
  <div>
    <label
      for="test"
    >
      test
    </label>
    <textarea
      id="test"
      name="test"
    />
  </div>
</div>
`;
